src/main/java/com/minecolonies/coremod/colony/Structures.java

```diff
@@ -60,200 +114,847 @@ public static void init()
-                    final String hut = filename.split("\\d+")[0];
+                    String relativePath = path.toString().substring(basePath.toString().length()).split("\\" + SCHEMATIC_EXTENSION)[0];
+                    if (!SCHEMATICS_SEPARATOR.equals(path.getFileSystem().getSeparator()))
+                    {
+                        relativePath = relativePath.replace(path.getFileSystem().getSeparator(), SCHEMATICS_SEPARATOR);
+                    }
+                    if (relativePath.startsWith(SCHEMATICS_SEPARATOR))
+                    {
+                        relativePath = relativePath.substring(1);
+                    }
 
-                    if (isSchematicHut(hut))
+                    final StructureName structureName = new StructureName(relativePath);
+                    final String md5 = Structure.calculateMD5(Structure.getStream(relativePath));
+                    if (md5 == null)
                     {
-                        addHutStyle(hut, style);
-                        incrementHutMaxLevel(hut);
+                        Log.getLogger().error("Structures: " + structureName + " with md5 null.");
                     }
-                    else
+                    else if (isSchematicSizeValid(structureName.toString()))
                     {
-                        addDecorationStyle(style, filename);
+                        if (md5Map.containsKey(structureName.toString()))
+                        {
+                            Log.getLogger().info("Override " + structureName + " md5:" + md5);
+                        }
+                        else
+                        {
+                            Log.getLogger().info("Add " + structureName + " md5:" + md5);
+                        }
+                        md5Map.put(structureName.toString(), md5);
+                        if (MineColonies.isClient())
+                        {
+                            addSchematic(structureName);
+                        }
                     }
                 }
             }
         }
+        catch (@NotNull IOException e)
+        {
+            Log.getLogger().warn("loadSchematicsForPrefix: Could not load schematics from " + basePath.resolve(prefix) + " (" + e.getMessage() +  ")");
+        }
     }
 
-    private static boolean isSchematicHut(final String name)
+    /**
+     * check that a schematic is not too big to be sent.
+     *
+     * @param structureName name of the structure to check for.
+     * @return True when the schematic is not too big.
+     */
+    private static boolean isSchematicSizeValid(@NotNull final String structureName)
     {
-        return Block.getBlockFromName(Constants.MOD_ID + ":blockHut" + name) != null;
+        final int maxSize = MAX_TOTAL_SIZE - Integer.SIZE / Byte.SIZE;
+        final byte[] data = Structure.getStreamAsByteArray(Structure.getStream(structureName.toString()));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') "structureName" is already a string, there's no need to call "toString()" on it. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1858)